include/gridtools/common/integral_constant.hpp

```diff
@@ -41,6 +41,19 @@ namespace gridtools {
         constexpr GT_FUNCTION operator T() const noexcept { return V; }
     };
 
+    // Returns the value_type of an integral_constant-like or returns the type itself
+    template <class T, class U = void>
+    struct to_integral_type {
+        using type = T;
+    };
+    template <class T>
+    struct to_integral_type<T, std::void_t<typename T::value_type>> {
+        using type = typename T::value_type;
+    };
```

Comment:
Here too, I would check for `is_integral_constant` which covers GT and std integral constants.

Reply:
done